ktor-utils/jvm/src/io/ktor/util/cio/Semaphore.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.util.cio
+
+import kotlinx.coroutines.sync.Semaphore
+
+@Deprecated(
+    "Ktor Semaphore is deprecated and will be removed in ktor 2.0.0. Consider using kotlinx.coroutines Semaphore instead.",
```

Comment:
Are we sure about 2.0.0? Isn't it better to no state exact version? I know we want to become more predictable. Are we going to do the same for other deprecations in the future?

Reply:
Yes, we are. We're going to remove deprecations only in major updates. I see no reason not to say it publicly.